supervisor/api/resolution.py

```diff
@@ -62,3 +78,17 @@ async def dismiss_issue(self, request: web.Request) -> None:
     def healthcheck(self, request: web.Request) -> Awaitable[None]:
         """Run backend healthcheck."""
         return asyncio.shield(self.sys_resolution.healthcheck())
+
+    @api_process
+    @require_home_assistant
+    async def check_options(self, request: web.Request) -> None:
+        """Set check options."""
+        body = await api_validate(SCHEMA_CHECK_OPTIONS, request)
+
+        try:
+            if body[ATTR_ENABLED]:
+                self.sys_resolution.check.enable(request.match_info.get("check"))
+            else:
```

Comment:
When `enabled` becomes `optional`, you should check here that it's not `None`, or else not providing `enabled` will disable it.